datalad/core/distributed/push.py

```diff
@@ -401,6 +401,20 @@ def _datasets_since_(dataset, since, paths, recursive, recursion_limit):
         yield (cur_ds, ds_res)
 
 
+def _transfer_data(repo, ds, target, content, data, force, jobs, res_kwargs,
```

Comment:
please add a comment or docstring for thing function then on its purpose in life, or the next PR from someone could be "collapsing not really necessary function".

But altogether -- such monkey patching from extensions is opening a precedence, or are we doing it already in other spots?  

We should at least come up with some consistent annotation so we could `git grep` for such locations: let's add a "do nothing" decorator `@todo_interface_for_extensions` to annotate such methods.  Could come handy later to identify such hidden treasures.

Reply:
It was introduced by @mih in `datalad-next` and was already used to patch the method `_set_request_headers` of `datalad.distributed.create_sibling_ghlike._GitHubLike`.

There is a discussion of the approach in another issue that I do not find yet. The monkey-patching approach is new and we have to see what can of worms it opens. The decorator is definitely a good idea! I will add it